homeassistant/components/mqtt/__init__.py

```diff
@@ -185,7 +186,13 @@ def valid_discovery_topic(value):
 }, required=True)
```

Comment:
paho-mqtt accepts `str`, `bytes`, `int`, `float` and `None` as payload. We don't check this here (`object`). Shall I write a validation function/"coerce" thing?